### PR TITLE
add TSSB-3M dataset

### PR DESCRIPTION
**Pull request by zirui**
add the TSSM-3M code bugs dataset
issue: [#1395](https://github.com/LAION-AI/Open-Assistant/issues/1395)

### CODE REVIEW

File: data/datasets/TSSB-3M/generate_dataset.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    """
+    # remove issue id , eg. msg: "rename (hetr_passes -> passes) #1195" -> "rename (hetr_passes -> passes)"
+    # remove `fix` prefix:
+    ##  eg. [fix] 拼写错误 -> 拼写错误
+    ## eg. [FIX] purchase_indonesia : AttributeError 'NoneType' object has no attribute 'id' ->  AttributeError 'NoneType' object has no attribute 'id'
+    """
+    # Remove issue id
+    # pattern = "#\d{1,6}|[fix]|[FIX]"
+    pattern = r"#\d{1,6}"
+    text = re.sub(pattern, "", text)
```
Comment (sedthh) [inline data/datasets/TSSB-3M/generate_dataset.py:79]: is this safe? wouldn't something like 
`re.sub(r"(.+?\s\(.+?\))\s#\d{1,6}", '\\1', text)`
work better?

File: data/datasets/TSSB-3M/README.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+bug fix is related to a commit in a public Python that does not change more than
+a single statement
+
+1. The original dataset comes from the
+   [TSSB-3M](https://zenodo.org/record/5845439) dataset
+2. By requesting the GitHub api to obtain the commit message, we expand and
+   create a new dataset
+   [TSSB-3M-ext](https://huggingface.co/datasets/zirui3/TSSB-3M-ext)
+3. Convert `TSSB-3M-ext` into instruction form to form the
+   [TSSB-3M-instruction](https://huggingface.co/datasets/zirui3/TSSB-3M-instructions)
```
Comment (sedthh) [inline data/datasets/TSSB-3M/README.md:13]: for some reason the HF dataset preview does not work (could be a caching problem that will just disappear on its own though)

File: data/datasets/TSSB-3M/generate_dataset.py [hunk 3]
```diff
@@ -0,0 +1,133 @@
+"""Convert the source TSSB-3M  dataset to instruction data
+"""
+
+import json
+import re
+from os.path import join
+
+from tqdm import tqdm
+
+TEMPLATE = """User: Find the bug in the following code:
```
Comment (sedthh) [inline data/datasets/TSSB-3M/generate_dataset.py:10]: maybe we should add random synonyms for the prompts to enrich the data

**Review by sedthh — CHANGES_REQUESTED** [review]
Had a few questions, awesome job nevertheless! This will make a great addition to the instruciton datasets.

File: data/datasets/TSSB-3M/README.md (hunk 2 above)
Comment (zirui) [reply data/datasets/TSSB-3M/README.md:13]: This is due to a problem with the. gz format. The data has been updated and can now be viewed normally

**Review by olliestanley — CHANGES_REQUESTED** [review]
The `.tsv` data file will need removing before merge

**Comment by zirui** [comment]
> The `.tsv` data file will need removing before merge
The. tsv data file (generated through statistics and review) is read in the script to filter commit messages

File: data/datasets/TSSB-3M/generate_dataset.py (hunk 3 above)
Comment (zirui) [reply data/datasets/TSSB-3M/generate_dataset.py]: > maybe we should add random synonyms for the prompts to enrich the data

Thank you for your suggestion. It has been modified to randomly generate synonymous prompts(instructions) from a  list

**Review by olliestanley — APPROVED** [review]
LGTM

File: data/datasets/__init__.py
Comment (0x22almostEvil) [inline data/datasets/__init__.py:1]: Can you fix a merge conflict by adding here:
    ```python 
    "poetry_instruction": "checkai/instruction-poems",
    "oa_stackexchange": "donfu/oa-stackexchange",
    "stable_diffusion_instructional_dataset": "MadVoyager/stable_diffusion_instructional_dataset",
    ```

**Review by 0x22almostEvil — CHANGES_REQUESTED** [review]

File: data/datasets/__init__.py
Comment (0x22almostEvil) [inline data/datasets/__init__.py:1]: Can you fix a merge conflict by adding here:

```python 
"poetry_instruction": "checkai/instruction-poems",
"oa_stackexchange": "donfu/oa-stackexchange",
"stable_diffusion_instructional_dataset": "MadVoyager/stable_diffusion_instructional_dataset",
```

**Review by 0x22almostEvil — CHANGES_REQUESTED** [review]

**Review by 0x22almostEvil — APPROVED** [review]

**Review by sedthh — APPROVED** [review]

**Pull request merged**